### PR TITLE
[SPARK-47258][SQL] Assign names to error classes _LEGACY_ERROR_TEMP_127[0-5]

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This PR renames a few error classes related to usage of SHOW CREATE TABLE errors:

_LEGACY_ERROR_TEMP_1270 => UNSUPPORTED_SHOW_CREATE_TABLE.ON_TEMPORARY_VIEW

_LEGACY_ERROR_TEMP_1271 => UNSUPPORTED_SHOW_CREATE_TABLE.WITH_UNSUPPORTED_FEATURE

_LEGACY_ERROR_TEMP_1272 => UNSUPPORTED_SHOW_CREATE_TABLE.ON_TRANSACTIONAL_HIVE_TABLE

_LEGACY_ERROR_TEMP_1273 => UNSUPPORTED_SHOW_CREATE_TABLE.WITH_UNSUPPORTED_SERDE_CONFIGURATION

_LEGACY_ERROR_TEMP_1274 => UNSUPPORTED_SHOW_CREATE_TABLE.ON_SPARK_DATA_SOURCE_TABLE_WITH_AS_SERDE

_LEGACY_ERROR_TEMP_1275 => UNSUPPORTED_SHOW_CREATE_TABLE.WITH_UNSUPPORTED_FEATURE

Also, this PR changes tests in corresponding test suite to use checkError() method which checks the error class name, context, error message parameters, etc.

### Why are the changes needed?

Proper error names and messages improve user experience with Spark SQL.

### Does this PR introduce _any_ user-facing change?

Yes, this PR changes user-facing error class and message.

### How was this patch tested?

By running tests from `ShowCreateTableSuiteBase` and subclasses.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @panbingkun @MaxGekk @itholic @beliefer

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 1]
```diff
@@ -2895,29 +2895,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
 
   def showCreateTableNotSupportedOnTempView(table: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1270",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.ON_TEMPORARY_VIEW",
```
Comment (wayneguow) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2898]: There is no `ON_TEMPORARY_VIEW` related test case because in `AstBuilder`, `allowTempView`parameter is set to `false` in the `visitShowCreateTable` method, and the code in `Analyzer`'s `ResolveRelations` rule `case u @ UnresolvedTableOrView(identifier, cmd, allowTempView)` will throw an `EXPECT_PERMANENT_VIEW_NOT_TEMP` exception.

File: common/utils/src/main/resources/error/error-conditions.json [hunk 2]
```diff
@@ -5923,40 +5923,40 @@
...
-  "_LEGACY_ERROR_TEMP_1270" : {
-    "message" : [
-      "SHOW CREATE TABLE is not supported on a temporary view: <table>."
-    ]
-  },
-  "_LEGACY_ERROR_TEMP_1271" : {
-    "message" : [
-      "Failed to execute SHOW CREATE TABLE against table <table>, which is created by Hive and uses the following unsupported feature(s)",
-      "<unsupportedFeatures>",
-      "Please use `SHOW CREATE TABLE <table> AS SERDE` to show Hive DDL instead."
```
Comment (wayneguow) [inline common/utils/src/main/resources/error/error-conditions.json:5935]: Whether `AS SERDE` is specified or not, an exception will be thrown, such as in the test case `hive partitioned view is not supported` of `HiveSQLViewSuite`, so this line is unnecessary, and merged `_LEGACY_ERROR_TEMP_1271 ` and `_LEGACY_ERROR_TEMP_1275 `.

File: common/utils/src/main/resources/error/error-conditions.json [hunk 3]
```diff
@@ -4673,6 +4673,41 @@
     },
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_SHOW_CREATE_TABLE" : {
```
Comment (panbingkun) [inline common/utils/src/main/resources/error/error-conditions.json:4676]: Can we reuse `FORBIDDEN_OPERATION` ?

File: common/utils/src/main/resources/error/error-conditions.json [hunk 4]
```diff
@@ -8240,4 +8240,4 @@
     ],
     "sqlState" : "P0001"
   }
-}
+}
```
Comment (panbingkun) [inline common/utils/src/main/resources/error/error-conditions.json:8243]: Please keep the end of `whitespace` in the original file.

File: common/utils/src/main/resources/error/error-conditions.json (hunk 3 above)
Comment (wayneguow) [reply common/utils/src/main/resources/error/error-conditions.json:4676]: @panbingkun It looks good, but I think that additional information needs to be told to the user for `WITH_UNSUPPORTED_FEATURE` and `WITH_UNSUPPORTED_SERDE_CONFIGURATION` error cases, should we use subclasses with more message under `FORBIDDEN_OPERATION`? and it will also bring some changes to the previous `FORBIDDEN_OPERATION` error.

**Comment by LuciferYang** [comment]
Is this one ready to go ? @panbingkun

**Comment by wayneguow** [comment]
> Is this one ready to go ? @panbingkun

After a discussion with @LuciferYang , we think that in `_LEGACY_ERROR_TEMP_1271 ` and `_LEGACY_ERROR_TEMP_1273 `, it is necessary to retain additional parameters and expose them to users. How about keeping the current ones and giving up reusing `FORBIDDEN_OPERATION`?

File: common/utils/src/main/resources/error/error-conditions.json [hunk 5]
```diff
@@ -4673,6 +4673,41 @@
     },
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_SHOW_CREATE_TABLE" : {
+    "message" : [
+      "Unsupported show create table command:"
+    ],
+    "subClass" : {
+      "ON_SPARK_DATA_SOURCE_TABLE_WITH_AS_SERDE" : {
```
Comment (wayneguow) [inline common/utils/src/main/resources/error/error-conditions.json:4681]: @panbingkun  I changed this error class to a more accurate name and to avoid ambiguity with `WITH_UNSUPPORTED_SERDE_CONFIGURATION`.

File: common/utils/src/main/resources/error/error-conditions.json [hunk 6]
```diff
@@ -4673,6 +4673,41 @@
...
+          "SHOW CREATE TABLE doesn't support transactional Hive table. Please use `SHOW CREATE TABLE <table> AS SERDE` to show Hive DDL instead."
+        ]
+      },
+      "WITH_UNSUPPORTED_FEATURE" : {
+        "message" : [
+          "Failed to execute SHOW CREATE TABLE against table/view <table>, which is created by Hive and uses the following unsupported feature(s)",
+          "<unsupportedFeatures>"
+        ]
+      },
+      "WITH_UNSUPPORTED_SERDE_CONFIGURATION" : {
```
Comment (LuciferYang) [inline common/utils/src/main/resources/error/error-conditions.json:4702]: I think 'WITH_UNSUPPORTED_CONFIG' may be more general, not necessarily limited to seder config, right?

On the other hand, I'm not sure if it's suitable to further merge WITH_UNSUPPORTED_FEATURE`` and `WITH_UNSUPPORTED_SERDE_CONFIGURATION` to make it more general."

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 7]
```diff
@@ -2895,29 +2895,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
...
   def showCreateTableNotSupportTransactionalHiveTableError(table: CatalogTable): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1272",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.ON_TRANSACTIONAL_HIVE_TABLE",
       messageParameters = Map("table" -> table.identifier.toString))
   }
 
   def showCreateTableFailToExecuteUnsupportedConfError(
       table: TableIdentifier,
       builder: mutable.StringBuilder): Throwable = {
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2910]: Unrelated to this PR, but passing in a `StringBuilder` makes me feel a bit strange.

File: common/utils/src/main/resources/error/error-conditions.json (hunk 5 above)
Comment (LuciferYang) [inline common/utils/src/main/resources/error/error-conditions.json:4681]: `ON_DATA_SOURCE_TABLE_WITH_AS_SERDE`?

File: common/utils/src/main/resources/error/error-conditions.json [hunk 8]
```diff
@@ -4673,6 +4673,41 @@
...
+          "<table> is a Spark data source table. Use `SHOW CREATE TABLE` without `AS SERDE` instead."
+        ]
+      },
+      "ON_TEMPORARY_VIEW" : {
+        "message" : [
+          "SHOW CREATE TABLE is not supported on a temporary view: <table>."
+        ]
+      },
+      "ON_TRANSACTIONAL_HIVE_TABLE" : {
+        "message" : [
```
Comment (LuciferYang) [inline common/utils/src/main/resources/error/error-conditions.json:4692]: I suggest making the message for `ON_TRANSACTIONAL_HIVE_TABLE` and `ON_SPARK_DATA_SOURCE_TABLE_WITH_AS_SERDE` more consistent in the pattern.

File: common/utils/src/main/resources/error/error-conditions.json (hunk 3 above)
Comment (LuciferYang) [reply common/utils/src/main/resources/error/error-conditions.json:4676]: > Can we reuse `FORBIDDEN_OPERATION` ?

Perhaps we can add a `FORBIDDEN_OPERATION_WITH_REASON` as well? WDYT @panbingkun @wayneguow ?

File: common/utils/src/main/resources/error/error-states.json [hunk 9]
```diff
@@ -2682,7 +2682,7 @@
         "usedBy": ["SQL/Foundation"]
     },
     "33000": {
-        "description": "invalid SQL descriptor nameno subclass)",
+        "description": "invalid SQL descriptor nameno subclass",
```
Comment (LuciferYang) [inline common/utils/src/main/resources/error/error-states.json:2685]: Is this change related to the current pr?

Comment (wayneguow) [reply common/utils/src/main/resources/error/error-states.json:2685]: It's not relevant, it's a small typo, just fixed.

Comment (wayneguow) [reply common/utils/src/main/resources/error/error-states.json:2685]: Sounds good, let me have a full check and separate a new PR.

**Review by MaxGekk — COMMENTED** [review]
@wayneguow Please, resolve conflicts and rebase on the recent master.

**Comment by wayneguow** [comment]
> @wayneguow Please, resolve conflicts and rebase on the recent master.

@MaxGekk I rebased the master branch and we can move it forward.

**Comment by wayneguow** [comment]
Gentle ping @MaxGekk , PTAL when you have time.

File: common/utils/src/main/resources/error/error-conditions.json [hunk 10]
```diff
@@ -5010,6 +5010,43 @@
...
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_SHOW_CREATE_TABLE" : {
+    "message" : [
+      "Unsupported show create table command."
+    ],
+    "subClass" : {
+      "ON_DATA_SOURCE_TABLE_WITH_AS_SERDE" : {
+        "message" : [
+          "<table> is a Spark data source table. Please use `SHOW CREATE TABLE` without `AS SERDE` instead."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5020]: Please, follow to `toSQLStmt()`, and just upper case the SQL statements:
```suggestion
          "The table <tableName> is a Spark data source table. Please use SHOW CREATE TABLE without AS SERDE instead."
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 11]
```diff
@@ -5010,6 +5010,43 @@
     },
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_SHOW_CREATE_TABLE" : {
+    "message" : [
+      "Unsupported show create table command."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5015]:
```suggestion
      "Unsupported a SHOW CREATE TABLE command."
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 12]
```diff
@@ -5010,6 +5010,43 @@
...
+    ],
+    "subClass" : {
+      "ON_DATA_SOURCE_TABLE_WITH_AS_SERDE" : {
+        "message" : [
+          "<table> is a Spark data source table. Please use `SHOW CREATE TABLE` without `AS SERDE` instead."
+        ]
+      },
+      "ON_TEMPORARY_VIEW" : {
+        "message" : [
+          "`SHOW CREATE TABLE` is not supported on a temporary view <table>."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5025]: You already mentioned `SHOW CREATE TABLE` in parent's message. Just refer to it like:
```suggestion
          "The command is not supported on a temporary view <table>."
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 13]
```diff
@@ -5010,6 +5010,43 @@
...
+        ]
+      },
+      "ON_TRANSACTIONAL_HIVE_TABLE" : {
+        "message" : [
+          "Failed to execute `SHOW CREATE TABLE` against transactional Hive table <table>.",
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5030]:
```suggestion
          "Failed to execute the command against transactional Hive table <tableName>.",
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 14]
```diff
@@ -5010,6 +5010,43 @@
...
+          "Please use `SHOW CREATE TABLE <table> AS SERDE` to show Hive DDL instead."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5031]:
```suggestion
          "Please use SHOW CREATE TABLE <tableName> AS SERDE to show Hive DDL instead."
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 15]
```diff
@@ -5010,6 +5010,43 @@
...
+        ]
+      },
+      "WITH_UNSUPPORTED_FEATURE" : {
+        "message" : [
+          "Failed to execute `SHOW CREATE TABLE` against table/view <table>, which is created by Hive and uses the following unsupported features",
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5036]:
```suggestion
          "Failed to execute the command against table/view <tableName> which is created by Hive and uses the following unsupported features",
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 16]
```diff
@@ -5010,6 +5010,43 @@
...
+          "<unsupportedFeatures>"
+        ]
+      },
+      "WITH_UNSUPPORTED_SERDE_CONFIGURATION" : {
+        "message" : [
+          "Failed to execute `SHOW CREATE TABLE` against table <table>, which is created by Hive and uses the following unsupported serde configuration",
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5042]:
```suggestion
          "Failed to execute the command against the table <tableName> which is created by Hive and uses the following unsupported serde configuration",
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 17]
```diff
@@ -5010,6 +5010,43 @@
...
+          "<configs>",
+          "Please use `SHOW CREATE TABLE <table> AS SERDE` to show Hive DDL instead."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5044]:
```suggestion
          "Please use SHOW CREATE TABLE <tableName> AS SERDE to show Hive DDL instead."
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 18]
```diff
@@ -2975,29 +2975,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
...
-      messageParameters = Map(
-        "unsupportedFeatures" -> table.unsupportedFeatures.map(" - " + _).mkString("\n"),
-        "table" -> table.identifier.toString))
-  }
-
   def showCreateTableNotSupportTransactionalHiveTableError(table: CatalogTable): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1272",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.ON_TRANSACTIONAL_HIVE_TABLE",
       messageParameters = Map("table" -> table.identifier.toString))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2985]: quote the table id by `toSQLId()`, please.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 19]
```diff
@@ -2975,29 +2975,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
...
   }
 
   def showCreateTableFailToExecuteUnsupportedConfError(
       table: TableIdentifier,
       builder: mutable.StringBuilder): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1273",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.WITH_UNSUPPORTED_SERDE_CONFIGURATION",
       messageParameters = Map(
         "table" -> table.identifier,
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2994]: Please, use `toSQLId()`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 20]
```diff
@@ -3006,18 +2998,18 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
   def showCreateTableAsSerdeNotAllowedOnSparkDataSourceTableError(
       table: TableIdentifier): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1274",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.ON_DATA_SOURCE_TABLE_WITH_AS_SERDE",
       messageParameters = Map("table" -> table.toString))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:3002]: Wrap by `toSQLId()`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 21]
```diff
@@ -3006,18 +2998,18 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
...
 
   def showCreateTableOrViewFailToExecuteUnsupportedFeatureError(
       table: CatalogTable,
-      features: Seq[String]): Throwable = {
+      unsupportedFeatures: Seq[String]): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1275",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.WITH_UNSUPPORTED_FEATURE",
       messageParameters = Map(
         "table" -> table.identifier.toString,
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:3011]: quote by `toSQLId()`

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 22]
```diff
@@ -2976,30 +2974,30 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
...
+        "tableName" -> toSQLId(table.nameParts),
+        "configs" -> configs))
   }
 
   def showCreateTableAsSerdeNotAllowedOnSparkDataSourceTableError(
       table: TableIdentifier): Throwable = {
     new AnalysisException(
       errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.ON_DATA_SOURCE_TABLE_WITH_AS_SERDE",
-      messageParameters = Map("table" -> table.toString))
+      messageParameters = Map("tableName" -> toSQLId(table.nameParts)))
```
Comment (wayneguow) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2984]: The usage of `identifier.nameParts` is common in `QueryCompilationErrors`, so I unified it in this way

**Comment by wayneguow** [comment]
@MaxGekk  Thanks for reviews. I updated as suggested.

**Review by MaxGekk — APPROVED** [review]
